### PR TITLE
Jmx for pcml connector for EI 6.5.0

### PR DESCRIPTION
**Pull request by markelari**

### CODE REVIEW

File: src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package org.wso2.carbon.connector.pcml;
```
Comment (SanojPunchihewa) [inline src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java:1]: Shall we add the license headers? Please add them to `PCMLPoolMBean` class as well

File: src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    int maxLifetime;
+    int maxUseCount;
+    int maxUseTime;
+    boolean isRunMaintenance;
+    boolean isThreadUsed;
+    int cleanupInterval;
+    boolean pretestConnections;
+
+
+    public PCMLPool(AS400ConnectionPool connectionPool) {
```
Comment (SanojPunchihewa) [inline src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java:19]: Do we have any method in the `AS400ConnectionPool` to get the currently active connections?

File: src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java (hunk 1 above)
Comment (markelari) [reply src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java:1]: done

File: src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java (hunk 2 above)
Comment (markelari) [reply src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java:19]: searched but did not find it

Comment (SanojPunchihewa) [reply src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java:19]: Shall we check [getActiveConnectionCount](https://www.ibm.com/docs/api/v1/content/ssw_ibm_i_74/rzahh/javadoc/com/ibm/as400/access/AS400ConnectionPool.html#getActiveConnectionCount(java.lang.String,%20java.lang.String))(java.lang.String systemName, java.lang.String userID) and [getAvailableConnectionCount](https://www.ibm.com/docs/api/v1/content/ssw_ibm_i_74/rzahh/javadoc/com/ibm/as400/access/AS400ConnectionPool.html#getAvailableConnectionCount(java.lang.String,%20java.lang.String))(java.lang.String systemName, java.lang.String userID) ?

File: src/main/java/org/wso2/carbon/connector/pcml/PCMLPoolMBean.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    long getMaxInactivity();
+    long getMaxLifetime();
+    int getMaxUseCount();
+    long getMaxUseTime();
+    boolean isRunMaintenance();
+    boolean isThreadUsed();
+    long getCleanupInterval();
+    boolean isPretestConnections();
+
+}
```
Comment (SanojPunchihewa) [inline src/main/java/org/wso2/carbon/connector/pcml/PCMLPoolMBean.java:15]: Shall we add a new line here at the end of the file

File: src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        this.as400ConnectionPool = connectionPool;
+    }
+
+    @Override
+    public int getMaxConnections() {
+        return this.getAs400ConnectionPool().getMaxConnections();
+    }
+
+    public AS400ConnectionPool getAs400ConnectionPool() {
```
Comment (SanojPunchihewa) [inline src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java:28]: Let's make this method private or use `this.as400ConnectionPool` in the rest of the methods

File: src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    int maxConnections;
+    long maxInactivity;
+    int maxLifetime;
+    int maxUseCount;
+    int maxUseTime;
+    boolean isRunMaintenance;
+    boolean isThreadUsed;
+    int cleanupInterval;
+    boolean pretestConnections;
+
```
Comment (SanojPunchihewa) [inline src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java:17]: Shall we remove these unnecessary empty lines

File: src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java (hunk 2 above)
Comment (SanojPunchihewa) [reply src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java:19]: @markelari, seems like you have missed pushing the changes

**Comment by arunans23** [comment]
Thank you for your contribution. This support is added with https://github.com/wso2-extensions/esb-connector-pcml/pull/27. Hence closing this PR.

**Pull request closed without merging**